packages/baseclient-components/src/container/LayerLegendAccordion/LayerLegendAccordion.tsx

```diff
@@ -0,0 +1,496 @@
+      const wmsSource: any = layer.getSource();
+      const encodedLayerNames = wmsSource.getParams().LAYERS;
+      loading = loadingQueue!.find((e: string) => e.indexOf(encodedLayerNames) > -1);
+    }
+    const loadingSpanClass = `fa fa-spinner fa-spin layer-tree-node-loading-${loading ? 'active' : 'inactive'}`;
+
+    return (
+      <span
+        className="layer-tree-node-list-item"
+      >
+        <div className="layer-tree-node-title">
+        <Tooltip
+          title={t('LayerLegendAccordion.toggleVisibilityTooltipText')}
+          placement="right"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className={visibilitySpanClass}
+            onClick={() => this.onLayerTreeNodeVisibilityChange(layer)}
+          />
+        </Tooltip>
+        <Tooltip
+          title={layer.get('name')}
+          placement="top"
+          mouseEnterDelay={0.5}
+        >
+          <span
+            className="layer-tree-node-title-layername"
+          >
+            {layer.get('name')}
+          </span>
+        </Tooltip>
+        <span
+            className={loadingSpanClass}
+          />
+        {
+          !(layer instanceof OlLayerGroup) ?
+            <Dropdown
+              overlay={<div />} // TODO: make something more senseful here
+              placement="topLeft"
+              trigger={['click']}
+            >
+              <Tooltip
+                title={this.props.t('LayerLegendAccordion.layerSettingsTooltipText')}
+                placement="right"
+                mouseEnterDelay={0.5}
+              >
+                <span
+                  className="fa fa-cog layer-tree-node-title-settings"
+                />
+              </Tooltip>
+            </Dropdown> :
+            null
+          }
+          </div>
+          <LayerTransparencySlider
+            layer={layer}
+          />
+        </span>
+    );
+  }
+
+  /**
+   * Generate legend items for currently visible layers on map
+   *
+   * @returns {Collapse[]} An array of collapsible Instances of <Legend>
+   */
+  getLegendItems() {
+    const { map } = this.props;
+    const layers = map.getLayerGroup().getLayers().getArray();
+    const scale = MapUtil.getScaleForResolution(map.getView().getResolution(), 'm');
+
+    // clone the array, reverse will work in place
+    const reversed = layers.slice(0).reverse().filter((l: any) => l.getVisible());
+    const legends = reversed.map((l: any) => {
+      return <Collapse
+        bordered={false}
+        destroyInactivePanel
+      >
+        <Panel
+          key={_uniqueId('panel-')}
+          header={l.get('name')}
+        >
+          <Legend
+            key={l.getRevision()}
+            layer={l}
+            extraParams={{
+              scale: scale,
+              WIDTH: 30 * 1.5,
+              HEIGHT: 30,
+              TRANSPARENT: true,
+              LEGEND_OPTIONS: 'fontAntiAliasing:true;forceLabels:on;fontName:DejaVu Sans Condensed'
```

Comment:
I would suggest to have this configurable as it's highly dependent on the service provider.

Reply:
+1

Reply:
Alright, moved `legend_options` to a new prop called `extraLegendParams`